From d1417aed0b335d03c67c998b17b72933764ccbe3 Mon Sep 17 00:00:00 2001
From: Matt Borgerson <contact@mborgerson.com>
Date: Mon, 23 Jan 2023 03:06:46 -0700
Subject: [PATCH] Rename pingtest's slirp_inet_aton

---
 test/pingtest.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/pingtest.c b/test/pingtest.c
index 3bb0488..04b63ce 100644
--- a/test/pingtest.c
+++ b/test/pingtest.c
@@ -24,7 +24,7 @@
 #ifdef _WIN32
 //#include <sys/select.h>
 #include <winsock2.h>
-int slirp_inet_aton(const char *cp, struct in_addr *ia)
+int pingtest_inet_aton(const char *cp, struct in_addr *ia)
 {
     uint32_t addr = inet_addr(cp);
     if (addr == 0xffffffff) {
@@ -33,7 +33,7 @@ int slirp_inet_aton(const char *cp, struct in_addr *ia)
     ia->s_addr = addr;
     return 1;
 }
-#define inet_aton slirp_inet_aton
+#define inet_aton pingtest_inet_aton
 #else
 #include <poll.h>
 #endif
-- 
2.34.1

